build/extract.js

```diff
@@ -0,0 +1,339 @@
+      let chars = this.extractAlphabet(alphabet);
+      if (chars.length) {
+        chars.forEach(char => {
+          // some languages (e.g. "BS") include character clusters
+          // "[a b c č ć d {dž} ... {lj} m n {nj} ... š t u v z ž]"
+          if (char && !/^{.+}$/.test(char)) {
+            // convert character into base letter + combining
+            // diacritic then remove any combining diacritics
+            let base = stripMarks(char.normalize("NFD"));
+            if (base !== char && base !== "") {
+              lang.data[char] = {
+                mapping: {
+                  base: base
+                }
+              };
+            }
+          }
+        });
+        this._results[language] = lang;
+      }
+    }
+  }
+
+  /**
+   * Validate language results
+   * - remove variants that exactly match the root
+   * - write language file to "src" directory
+   */
+  validateList() {
+    const languages = Object.keys(this._results);
+    // assuming the first language listed isn't a variant
+    let root = JSON.stringify(this._results[languages[0]]);
+    languages.forEach(language => {
+      let unique = true,
+        isVariant = /-/.test(language) &&
+          // make exception for "sr-Latn"
+          !/^sr-Latn$/.test(language);
+      const data = this._results[language];
+      if (!isVariant) {
+        root = JSON.stringify(data);
+      }
+      // skip validated languages
+      if (!this._validLangs.includes(language) ||
+        isVariant &&
+        !this._validLangs.includes(this.getRootLang(language))
+      ) {
+        // target language variants
+        if (isVariant) {
+          // include variants that are different from the root
+          // language
+          if (JSON.stringify(data) === root) {
+            unique = false;
+          }
+        }
+        // write file if unique and it contains data
+        if (unique && Object.keys(data.data).length) {
+          this.writeOutput(language, data);
+        }
+      }
+    });
+  }
+
+  /**
+   * Writes the defined content into ./src/[lang]/[lang].json
+   * @param {string} language - IETF Language tag
+   * @param {object} data - Language JSON data to output
+   */
+  writeOutput(language, data) {
+    let indx, temp,
+      folder = language;
+    if (/-/.test(language)) {
+      // per spec... de/ would contain de.js, at.js & ch.js
+      indx = language.indexOf("-");
+      folder = language.substring(0, indx);
+      // make exception for "sr-Latn"
+      language = /^sr-Latn$/.test(language) ?
+        folder :
+        language.slice(indx + 1);
+    }
+    temp = `./src/${folder}/`;
+    if (!fs.existsSync(temp)) {
+      fs.mkdirSync(temp);
+    }
+    temp = `./src/${folder}/${language}.js`;
+    if (!fs.existsSync(temp)) {
+      fs.writeFileSync(
+        temp,
+        unvalidatedMessage + JSON.stringify(data, null, 4) + "\n",
+        "utf8"
+      );
+    } else if (language !== "sr") {
```

Comment:
The [Serbian ("sr") language](https://en.wikipedia.org/wiki/Serbian_language) has two alphabets, Cyrillic and Latin. The Cyrillic alphabet, as far as I can tell, doesn't have any diacritics. but the Latin version does...

If I remember correctly, this code prevents the "sr" file from being overwritten because the "sr" folder contains the Cyrillic alphabet while the "sr-Latn" folder has the Latin alphabet which contains diacritics.

I probably should rewrite this code to make it less fragile. I don't know when I'll be able to get to it, but since the "sr" language is already in the "validated-languages.json", it'll get ignored during the extract process.